src/datablocks/datablock_limit.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 The VectorSQL Authors.
+//
+// Code is licensed under Apache License, Version 2.0.
+
+package datablocks
+
+func (block *DataBlock) Limit(offset, limit int) (*DataBlock, error) {
+	result := block.Clone()
+
+	rows := block.NumRows()
+	if rows <= offset {
+		return result, nil
+	}
+	end := limit + offset
+	if end > rows {
+		end = rows
+	}
+
+	for i := range block.values {
```

Comment:
We can't range the block.values directly, the values have order which matained by block.seqs.
Using iterator.Next() and iterator.Value() to get the right limit records.

Reply:
We can limit the block.seqs instead handle the block.values

Reply:
In datablocks, each row/column-value get must through the iterator not the block.values.